lib/store/ledger.ex

```diff
@@ -9,12 +9,85 @@ defmodule Elixium.Store.Ledger do
 
   @store_dir ".chaindata"
   @ets_name :chaindata
+  @block_cache :block_cache
 
   def initialize do
     initialize(@store_dir)
     :ets.new(@ets_name, [:ordered_set, :public, :named_table])
+    :ets.new(@block_cache, [:ordered_set, :bag, :named_table])
   end
 
+  @doc """
+    Store & Delete Blocks from the block cache ets table
+  """
+  def store(block) do
+    validate_operation(:ets.insert(@block_cache, {:"#{block.index}", block}))
+  end
+
+  def delete(block) do
+    validate_operation(:ets.delete(@block_cache, {block}))
+  end
+
+  def delete_block(block_index) do
+    validate_operation(:ets.delete(@block_cache, :"#{block_index}"))
+  end
+  @doc """
+    First we check if the store is empty, if the store is empty initially we have nothing to match with so lets store it, else lets check the index
+  """
+  def block_cache_validation(current_block) do
+    case check_cache_size do
+      0 ->
+        with current_block <- store(current_block) do
+          :ok
+        end
+      _->
+        check_index(current_block)
+    end
+  end
+
+  def check_cache_size do
+    :ets.info(@block_cache, :size)
+  end
+
+  @doc """
+    We know were looking for a matching partner in the table so +1 in this case
+  """
+  defp check_index(current_block) do
```

Comment:
The ledger should only contain functionality to manipulate the blocks in the chain -- any checking / validation functionality belongs outside of this module. Perhaps this belongs in the orphan pool module?